src/http/http.hpp

```diff
@@ -13,6 +13,8 @@
 #include <boost/shared_array.hpp>
 #include <boost/optional.hpp>
 
+#include <openssl/ssl.h>
```

Comment:
This should be at the top of all `#include`s in this file (before the boost includes and before the C++ STL includes)